tools/audit_policy/main.go

```diff
@@ -46,6 +47,75 @@ var auditPolicyBasicServiceAccounts = []string{
 }
 `
 
+func main() {
+	workDir := cwd()
+
+	var (
+		output string
+		stream = os.Stdout
+	)
+	flag.StringVar(&output, "output", "", "output file for generated code")
+	flag.Parse()
+
+	if output != "" {
+		var err error
+		stream, err = os.Create(output)
+		if err != nil {
+			panic(err)
+		}
+
+		defer stream.Close()
+	}
+
+	var namespacesMap = make(map[string]struct{})
+	var sasMap = make(map[string]struct{})
+
+	// 1. find all modules
+	res, err := findModules(workDir)
+	if err != nil {
+		panic(err)
+	}
+
+	// 2. find templates for each module and fill ServiceAccounts info
+	for moduleYamlPath, moduleDirName := range res {
+		err = processModule(workDir, moduleYamlPath, moduleDirName, sasMap, namespacesMap)
```

Comment:
I don't understand why we need to pass these two maps as arguments, judging by the code we don't use any caching mechanism between modules. Can we just return new maps via return values?